packages/client-search/src/methods/client/hasPendingMappings.ts

```diff
@@ -0,0 +1,25 @@
+import { MethodEnum } from '@algolia/requester-common';
+import { RequestOptions } from '@algolia/transporter';
+
+import { HasPendingMappingsOptions, HasPendingMappingsResponse, SearchClient } from '../..';
+
+export const hasPendingMappings = (base: SearchClient) => {
+  return (
+    requestOptions?: HasPendingMappingsOptions & RequestOptions
+  ): Readonly<Promise<HasPendingMappingsResponse>> => {
+    const { retrieveMappings, ...options } = requestOptions || {};
+
+    if (retrieveMappings === true) {
+      // eslint-disable-next-line functional/immutable-data
+      options.getClusters = true;
```

Comment:
I don't see any unit tests for this, on purpose?

Reply:
```
  await expect(client.hasPendingMappings({ retrieveMappings: true })).resolves.toMatchObject({
     clusters: expect.any(Object),
     pending: expect.any(Boolean),
   });
```

Reply:
that's an integration test, no?

Reply:
@Haroenv Indeed, but what is the point of having a unit test if the integration one already covers all the cases?